src/document.rs

```diff
@@ -548,7 +546,7 @@ impl Document {
 
     fn extend(
         &mut self,
-        py_dict: &PyDict,
+        py_dict: &Bound<PyDict>,
```

Comment:
This one takes `&Bound` instead of the many other functions that take `Bound`.  I did this because it makes the caller code more convenient.

Reply:
If it is sufficient, then `&Bound` should almost always be preferred as it borrows the Python reference instead of bumping the reference count and the double indirection is usually optimized away by the compiler.

Reply:
Ok that's useful info. I wasn't sure because the blog said that the new version of `&PyAny` was `Bound<...>` (without the &).  So I expected clones to be cheap, which they were. I was surprised that there wasn't an explicit recommendation of `&Bound` over `Bound`, but I suppose that's just normal Rust stuff and not worth calling out specifically.  I guess I was just too literal.

![image](https://github.com/quickwit-oss/tantivy-py/assets/480395/4d1fb4cc-e62c-4029-8720-f4d444ec0065)